### PR TITLE
Dmeerx 597 glucose monitor lab

### PR DESCRIPTION
**Pull request by ghost**
How to test:

Testing **WITH pre-population of lab results** (when patient does not have qualifying pre-existing condition):
            a. Ensure you have pulled down the `DMEERX-597-observation-glucose-labs` branch of test-ehr. 
           b. Run crd 
           c. In the request generator, select patient **Ada Wilson** and **Home Blood Glucose Monitor device request**.
            d. Click **Lab Form** on the CDS card that is returned.
            e. Expected result: Lab Form should be pre-populated with: 
               - Patient information
                - Practitioner who is providing face-to-face evaluation 
               - Test Results (initial AND confirmatory--though they will be the same as I only added a single lab result resource of each test) 
               - Provider signature (though the signature itself is not pre-populated as I assumed the Provider would have to do that manually) 

Testing **WITHOUT pre-population of lab results** (when patient does have qualifying pre-existing condition) : 
        a. Ensure you have pulled down the `DMEERX-597-condition-diabetes` branch of test-ehr.
         b-d. Same has above 
        e. Expected result: Lab Form should be pre-populated with:
               - Patient information 
              - Practitioner who is providing face-to-face evaluation 
              - Provider signature 

Few notes:
	•	Lab results are only pre-populated if the patient does NOT meet the qualifying requirements (i.e. the patient does NOT have a pre-existing condition related to glucose intolerance)
	•	Lab name/Tester name is not pre-populated. Happy to add if you think it's necessary.

### CODE REVIEW

File: HomeBloodGlucoseMonitor/R4/resources/Questionnaire-R4-HomeBloodGlucoseMonitorLab.json [hunk 1]
```diff
@@ -0,0 +1,631 @@
...
+            "code": "other",
+            "display": "Other"
+          }
+        ]
+      }
+    }
+  ],
+  "item": [
+    {
+      "linkId": "1",
```
Comment (yunwwang) [inline HomeBloodGlucoseMonitor/R4/resources/Questionnaire-R4-HomeBloodGlucoseMonitorLab.json:24]: Use Patient-Info subquestionnair

File: HomeBloodGlucoseMonitor/R4/resources/Questionnaire-R4-HomeBloodGlucoseMonitorLab.json [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+                "language": "text/cql",
+                "expression": "\"BasicPatientInfoPrepopulation\".MedicareId"
+              }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "linkId": "2",
```
Comment (yunwwang) [inline HomeBloodGlucoseMonitor/R4/resources/Questionnaire-R4-HomeBloodGlucoseMonitorLab.json:35]: Use Practitioner-Info subquestionnaire

File: HomeBloodGlucoseMonitor/R4/resources/Questionnaire-R4-HomeBloodGlucoseMonitorLab.json [hunk 3]
```diff
@@ -0,0 +1,631 @@
...
+              "required": true,
+              "linkId": "4.10",
+              "text": "A1c level [%]"
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "type": "group",
```
Comment (yunwwang) [inline HomeBloodGlucoseMonitor/R4/resources/Questionnaire-R4-HomeBloodGlucoseMonitorLab.json]: Use ProviderSignature sub questionnaire

File: HomeBloodGlucoseMonitor/R4/resources/Questionnaire-R4-HomeBloodGlucoseMonitorLab.json (hunk 1 above)
Comment (ghost) [reply HomeBloodGlucoseMonitor/R4/resources/Questionnaire-R4-HomeBloodGlucoseMonitorLab.json:24]: Will do

File: HomeBloodGlucoseMonitor/R4/resources/Questionnaire-R4-HomeBloodGlucoseMonitorLab.json (hunk 3 above)
Comment (ghost) [reply HomeBloodGlucoseMonitor/R4/resources/Questionnaire-R4-HomeBloodGlucoseMonitorLab.json]: Will do

File: HomeBloodGlucoseMonitor/R4/resources/Questionnaire-R4-HomeBloodGlucoseMonitorLab.json (hunk 2 above)
Comment (ghost) [reply HomeBloodGlucoseMonitor/R4/resources/Questionnaire-R4-HomeBloodGlucoseMonitorLab.json:35]: Will do

**Comment by lpdecelles** [comment]
image

I was not able to get this to work with the latest from master and the **DMEERX-597-glucose-monitor-lab** branch. See the above screen shot of Questionnaire trying to load.

**Comment by lpdecelles** [comment]
The Questionnaire loads now. I did not see test results. I am using the CDS-Library branch DMEERX-597-glucose-monir-lab. Are you seeing test results?   

image

**Comment by lpdecelles** [comment]
This works for "Ada Wilson", but not for "Vlad Quinton".

Is there no pre-fetch data available for "Vlad Quinton"?

**Review by yunwwang — APPROVED** [review]
Two subQuestionnaires looks good

**Comment by yuliumitre** [comment]
For lab results, we have discussed that we are going to show the Lab result, test date and optional comments as a standard way to display the results. We can address those issues in the lab results ticket.

**Review by yuliumitre — APPROVED** [review]

**Pull request merged**